win_installer.py

```diff
@@ -139,6 +139,29 @@ def about_dict(repo_root):
     return about
 
 
+def pypi_wheel_release(name, version):
+    """
+    Returns a `name==version` string if there's such a wheel at PyPI. When
+    version ends with ".0", a shorter version may be returned, matching what's
+    found at PyPI (name=`PyQtChart` version=`5.12.0` returns `PyQtChart==5.12`,
+    as of this writing). When no releases are found, a ValueError is raised.
+    """
+    package = yarg.get(name)
+    while True:
+        releases = package.release(version)
+        if releases or not version.endswith(".0"):
+            break
+        version = version[:-2]
+
+    if not releases:
+        raise ValueError("No {n!r} package found at PyPI.".format(n=name))
+
+    if any(r.package_type == "wheel" for r in releases):
+        return "{n}=={v}".format(n=name, v=version)
+
+    return None
```

Comment:
If we raise a valuerror here as well we can also simplify the code in `pypi_wheels_in`.

Reply:
Returning `None` is different from raising a `ValueError` here:
* `None` the requested package version exists, but there are no wheels at PyPI.
  This will have the dependency be added as a non-wheel dependency in `pynsist`'s config.
  AFAICT, `pynsist` will copy those dependencies directly from the environment into the
  packaged artifact -- this fails badly in the case of `PyQtChart`.
* `ValueError` no such package version exists at PyPI (ie: `PyQtChart==5.12.0`).
  This is the `PyQtChart` case we're hitting now: given the meta-data inconsistency,
  even though our `setup.py` requires `==5.12`, the packaging venv version comes out
  as `==5.12.0` (`pip freeze --all` output) which `yarg` fails to locate and, thus,
  fails to add to the `pynsist` config wheel list. Now here we have two options, I guess:
  * Print out a "BIG THIS LOOKS BAD" message and move on (having the dependency
     added as a non-wheel in `pynsist` config, and bound to fail later)
  * Print out a big "INCONSISTENT METADATA" message and abort right there.

What do you say?

Reply:
Yeah, I agree with your reasoning of using None in this case.
I generally try to avoid returning None unless it's very clear that is an option simply by reading the code, but that's what function docs are for as well, and in this case it's not that unexpected.

Reply:
> -   Print out a big "INCONSISTENT METADATA" message and abort right there.


This is my preferred option.

Reply:
I'll update the PR for that simpler, stricter behaviour. :)